withastro/astro
# Actions: New fallback behavior with `action={actions.name}`

## Changes

Introduce the new pattern `action={actions.name}` for form fallbacks. This replaces the previous `getActionProps()` pattern.

```astro
---
import { actions } from 'astro:actions';
---

<form action={actions.logOut}>
  <button>Log Out</button>
</form>
```

- Deprecate `getActionProps()`
- Allow actions to return a string
- Introduce `.queryString` as an action property for more explicit URL construction
- Update React 19 handler to also use a query string for form actions


## Testing

- Add integration tests for new pattern, marking old tests as legacy
- Check that e2e tests still pass

## Docs

RFC updates to be pushed on minor release
